mne/time_frequency/tfr.py

```diff
@@ -673,13 +682,32 @@ def plot(self, picks=None, baseline=None, mode='mean', tmin=None,
 
         tmin, tmax = times[0], times[-1]
 
-        for k, p in zip(range(len(data)), picks):
-            fig = plt.figure()
-            _imshow_tfr(plt, 0, tmin, tmax, vmin, vmax, ylim=None,
-                        tfr=data[k: k + 1], freq=freqs, x_label='Time (ms)',
-                        y_label='Frequency (Hz)', colorbar=colorbar,
-                        picker=False, cmap=cmap, title=title)
-
+        if axes is None:
+            for idx in range(len(data)):
+                fig = plt.figure()
+                _imshow_tfr(plt, 0, tmin, tmax, vmin, vmax, ylim=None,
+                            tfr=data[idx: idx + 1], freq=freqs,
+                            x_label='Time (ms)', y_label='Frequency (Hz)',
+                            colorbar=colorbar, picker=False, cmap=cmap,
+                            title=title)
+        else:
+            if isinstance(axes, plt.Axes):
+                axes = [axes]
+            if len(axes) != len(picks):
+                raise RuntimeError('There must be an axes for each picked ' +
+                                   'channel.')
+            if colorbar:
+                logger.warning('Cannot draw colorbar for user defined axes.')
+            for idx in range(len(data)):
+                fig = axes[idx].get_figure()
+                _imshow_tfr(axes[idx], 0, tmin, tmax, vmin, vmax, ylim=None,
+                            tfr=data[idx: idx + 1], freq=freqs,
+                            x_label='Time (ms)', y_label='Frequency (Hz)',
+                            colorbar=False, picker=False, cmap=cmap,
+                            title=title)
```

Comment:
This does not seem as DRY as it could be. My procedure would be to triage the `axes` argument first, either creating the axes (if `axes is None` or checking the input for correctness), then do a single unified plotting call.